src/lib/session.ts

```diff
@@ -0,0 +1,67 @@
+import * as router from "@app/lib/router";
+import { derived, writable } from "svelte/store";
+
+interface Session {
+  id: string;
+  publicKey: string;
+}
+const store = writable<Session | undefined>(undefined);
+export const sessionStore = derived(store, s => s);
+
+export async function create(params: {
+  id: string;
+  signature: string;
+  publicKey: string;
+}) {
+  disconnect();
+
+  const request = await fetch(
+    `http://0.0.0.0:8080/api/v1/sessions/${params.id}`,
+    {
+      method: "PUT",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ sig: params.signature, pk: params.publicKey }),
+    },
+  );
+  if (request.ok) {
+    save(params.id, params.publicKey);
+  }
+
+  router.push({ resource: "home" });
+}
+
+export function disconnect() {
+  window.localStorage.removeItem("session");
```

Comment:
Just FYI, this is not a review 😅
We have a session delete route now in `radicle-httpd`, which we should call when disconnecting the session.
https://github.com/radicle-dev/heartwood/blob/c138cd062c8251a67c0ca784251ae5d7503501b4/radicle-httpd/src/api/v1/sessions.rs#L29

Reply:
This is implemented now.